pdns/dnsrecords.cc

```diff
@@ -45,15 +45,26 @@ void DNSResourceRecord::setContent(const string &cont) {
 
 string DNSResourceRecord::getZoneRepresentation(bool noDot) const {
   ostringstream ret;
+  vector<string> parts;
+  string last;
+
   switch(qtype.getCode()) {
     case QType::SRV:
     case QType::MX:
+      stringtok(parts, content);
+      last = *parts.rbegin();
+      ret << content;
+      if (last == ".")
+        break;
+      if (*(last.rbegin()) != '.' && !noDot)
+        ret << ".";
+      break;
     case QType::CNAME:
     case QType::DNAME:
     case QType::NS:
     case QType::PTR:
-      if (*(content.rbegin()) != '.') {
-        ret<<content;
+      ret<<content;
+      if (content != ".") {
```

Comment:
I'm not sure I agree with this check, as it seems to me that it will add a dot if the existing content is not a single dot, even if it does already end with one? Perhaps the existing check makes more sense, as long as:
> res<<content;

is done before the check.

Reply:
If I feed in `15 example.com.` with noDot false, out comes `15 example.com.` so I think I'm missing the problem

Reply:
Sorry, you are right. Pushed another commit.